tasks/UnderwaterTask.cpp

```diff
@@ -29,20 +30,22 @@ bool UnderwaterTask::configureHook()
     node = transport::NodePtr(new transport::Node());
     node->Init();
     fluid_velocityPublisher = node->Advertise<gazebo::msgs::Vector3d>("/" + topicName);
-    gzmsg << "UnderwaterTask: advertising to gazebo topic ~/" + topicName << endl;
+    gzmsg << "UnderwaterTask: advertising to gazebo topic /" + topicName << endl;
     return true;
 }
 
 void UnderwaterTask::setGazeboModel(std::string const& pluginName, ModelPtr model)
 {
-    string worldName = GzGet((*(model->GetWorld())), Name, ());
+    topicName = std::regex_replace(pluginName, std::regex("__"), "/") + "/fluid_velocity";
```

Comment:
This calculation is also common. Move it in a base implementation in ModelPluginTask (e.g. `getTopicNameFromPluginName(pluginName)`)